dll/win32/lpk/lpk.c

```diff
@@ -65,19 +65,29 @@ LpkGetCharacterPlacement(
     DWORD dwUnused)
 {
     WORD *glyphs = NULL;
+    UINT *lpOrder = NULL;
+    LPWSTR lpOutString;
     INT cGlyphs;
 
     UNREFERENCED_PARAMETER(dwUnused);
     
-    /* Sanity check */
-    if( !(dwFlags & GCP_REORDER))
+    /* Sanity checks */
+    if ( !(dwFlags & GCP_REORDER))
        return FALSE;
-    
-    BIDI_Reorder(NULL, lpString, uCount, dwFlags, WINE_GCPW_FORCE_LTR, lpResults->lpOutString,
-                 lpResults->nGlyphs, lpResults->lpOrder, &glyphs, &cGlyphs);
+
+    lpOutString = HeapAlloc(GetProcessHeap(), 0, uCount * sizeof(WCHAR));
```

Comment:
Aren't you leaking lpOutString afterwards? (Also, you should certainly check whether the memory allocation succeeded or not).